pkg/v1/tkg/tkgpackageclient/package_update.go

```diff
@@ -55,31 +59,25 @@ func (p *pkgClient) UpdatePackage(o *tkgpackagedatamodel.PackageOptions, progres
 	}
 
 	pkgInstallToUpdate := pkgInstall.DeepCopy()
-	if o.Version != pkgInstallToUpdate.Status.Version {
-		if pkgInstallToUpdate.Spec.PackageRef == nil || pkgInstallToUpdate.Spec.PackageRef.VersionSelection == nil {
-			err = errors.New(fmt.Sprintf("failed to update package '%s'", o.PkgInstallName))
-			return
-		}
-		progress.ProgressMsg <- fmt.Sprintf("Getting package metadata for '%s'", pkgInstallToUpdate.Spec.PackageRef.RefName)
-		o.PackageName = pkgInstallToUpdate.Spec.PackageRef.RefName
-		if _, _, err = p.GetPackage(o); err != nil {
```

Comment:
According to https://vmware.slack.com/archives/GTNH9JL3S/p1633485806340900?thread_ts=1633480428.336400&cid=GTNH9JL3S, let's discuss this in tomorrow's sync up.

Reply:
When discussed, the decision was to have early-on validation rather than waiting for reconciliation message.